### ISSUE TITLE
Does not work with yarn.

### ISSUE THREAD

**Issue by romaia** [brombal/just-install#3]
Just tried using with yarn, but it fails with:

```
$ yarn foo
/home/romaia/dev/stoq/stoq-pdv/node_modules/just-install/bin/just:1
ELF
^

SyntaxError: Invalid or unexpected token
    at Object.compileFunction (node:vm:360:18)
    at wrapSafe (node:internal/modules/cjs/loader:1055:15)
    at Module._compile (node:internal/modules/cjs/loader:1090:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
```

This is after I added the script to my package.json. Also, trying to run directly with `yarn just` fails with the same error.

I see that the file from the error `/home/romaia/dev/stoq/stoq-pdv/node_modules/just-install/bin/just` is the actual just executable, and running it directly works, but that is not ideal.

I havent tried with npm/npx as that is not what I use.

**Comment by brombal** [comment]
Thanks for letting me know @romaia, I'll take a look. Can you let me know what OS & node version you are using?

**Comment by romaia** [comment]
Sorry for the delay.

```
romaia@stoqws-16:~/dev$ node --version
v16.17.0
romaia@stoqws-16:~/dev$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy

```

Also tried with `v18.12.1` and no luck

**Comment by brombal** [comment]
Hey @romaia, I wonder if you can try out the new release, 1.0.10. Just run `npm install just-install@1.0.10` in your project. I couldn't reproduce your exact problem, but I did notice some other issues yarn was having. I think was able to resolve them, so maybe your issue is fixed too.

**Comment by brombal** [comment]
If you're interested what the problem was, there is an outstanding issue with yarn (https://github.com/yarnpkg/yarn/issues/3421#issuecomment-443972019), where yarn is not installing binaries that are built as part of the install process. `just-install` runs just's pre-built binary install script as a `preinstall` script in package.json, and yarn wasn't linking the resulting binary file to `node_modules/.bin`. 

I worked around this by including a stub binary file that yarn can properly link, and then running just's installer as a `postinstall`, which replaces the stub file. It's a little hacky, but perhaps someday yarn will resolve this issue.

**Comment by romaia** [comment]
So. I tried version 1.0.10 and it still didn't work.

Trying to create a reproducible environment, I realized I am using yarn berry, and now I can reproduce it with a minimal installation.

First, you need to install yarn berry (Just tested with version 3.3.1), as instructed in https://yarnpkg.com/getting-started/install

After initializing an empty project, adding `just-install` as a dependency and trying to run it with `yarn just`, the same error I reported happens.

So in summary:

```
corepack enable
corepack prepare yarn@stable --activate
mkdir -p /tmp/foo && cd /tmp/foo 
yarn init -2
yarn add -ED just-install
yarn just
#<boom>
```

ps: I tested version 1.0.10 with a yarn 1.x version and it did in fact work.

**Comment by brombal** [comment]
Ah thanks, I was not using corepack to install (didn't realize it's no longer `npm i -g yarn` - that would explain why the old version worked for me). I was able to reproduce your error that way at least, so hopefully I should have a fix soon.

**Comment by brombal** [comment]
@romaia Ok, I think I have it working. Could you try it out again? 1.0.11 is the latest version.

**Comment by romaia** [comment]
I can confirm it works now. Thanks for the quick fix.
